spec-main/extensions-spec.ts

```diff
@@ -175,9 +177,62 @@ describe('chrome extensions', () => {
     });
   });
 
+  describe('chrome.webRequest', () => {
+    async function fetch(contents: WebContents, url: string) {
+      return contents.executeJavaScript(`fetch("${url}")`);
+    }
+
+    describe('chrome.webRequest.onBeforeRequest', () => {
+      it('can cancel the request', async () => {
+        const customSession = session.fromPartition(`persist:${uuid.v4()}`);
+        const w = new BrowserWindow({ show: false, webPreferences: { session: customSession, sandbox: true } });
+
+        w.webContents.once('dom-ready', async () => {
+          await customSession.loadExtension(path.join(fixtures, 'extensions', 'chrome-webRequest'));
+
+          await expect(fetch(w.webContents, url)).to.eventually.be.rejectedWith(TypeError);
+        });
+
+        w.loadURL(url);
+      });
+
+      it('does not cancel the request', async () => {
+        const customSession = session.fromPartition(`persist:${uuid.v4()}`);
+        const w = new BrowserWindow({ show: false, webPreferences: { session: customSession, sandbox: true } });
+
+        w.webContents.once('dom-ready', async () => {
+          await expect(fetch(w.webContents, url)).to.have.property('status', 200);
+        });
+
+        w.loadURL(url);
+      });
+    });
+
+    describe('chrome.webRequest.onBeforeRequest and Electron webRequest', () => {
+      it('chrome.webRequest takes precedence over Electron webRequest', async () => {
+        const customSession = session.fromPartition(`persist:${uuid.v4()}`);
+        const w = new BrowserWindow({ show: false, webPreferences: { session: customSession, sandbox: true } });
+
+        w.webContents.once('dom-ready', async () => {
+          await customSession.loadExtension(path.join(fixtures, 'extensions', 'chrome-webRequest'));
+
+          customSession.webRequest.onBeforeRequest((details: any, callback: any) => {
+            callback({
+              cancel: true
+            });
+          });
+
+          await expect(fetch(w.webContents, url)).to.eventually.be.rejectedWith(TypeError);
```

Comment:
Is there actually a better way to get the `net::ERR_BLOCKED_BY_CLIENT` error? `fetch` throws only a `TypeError` with `Failed to fetch` message.